kafka/kf_tools.py

```diff
@@ -11,6 +11,39 @@
 from utils import block_diag
 
 
+def band_selecta(band):
+    if band == 0:
+        return np.array([0, 1, 6, 2])
+    else:
+        return np.array([3, 4, 6, 5])
+
+
+def hessian_correction_pixel(gp, x0, C_obs_inv, innovation, band, nparams):
+    selecta = band_selecta(band)
+    ddH = gp.hessian(np.atleast_2d(x0[selecta]))
+    big_ddH = np.zeros((nparams,nparams))
+    for i, ii in enumerate(selecta):
+        for j, jj in enumerate(selecta):
+            big_ddH[ii,jj] = ddH.squeeze()[i,j]        
+    big_Hessian_corr = big_ddH*C_obs_inv*innovation
+    return big_Hessian_corr
+
+
+def hessian_correction(gp, x0,P_inv, innovation, mask, state_mask, band, nparams):
+    C_obs_inv = P_inv.diagonal()[state_mask.flatten()]
```

Comment:
Hmmm... I think this isn't correct, as P_inv does have the prior term and the observational term (which is the one you're after). I think what is needed might be in the observations object, in field `uncertainty`, but I need to check this

Reply:
Ah, yes, I think you are right. I'll look into it.

Reply:
Actually - I do pass in R_mat - the observation uncertainty. So the code is correct just with poorly named variables. I will fix that.